src/rules/pyupgrade/rules/f_strings.rs

```diff
@@ -0,0 +1,319 @@
+use rustpython_ast::{Constant, Expr, ExprKind, KeywordData};
+
+use crate::ast::types::Range;
+use crate::checkers::ast::Checker;
+use crate::fix::Fix;
+use crate::registry::Diagnostic;
+use crate::rules::pyflakes::format::FormatSummary;
+use crate::violations;
+
+// Checks for curly brackets. Inside these brackets this checks for an optional
+// valid python name and any format specifiers afterwards
+static NAME_SPECIFIER: Lazy<Regex> =
+    Lazy::new(|| Regex::new(r"\{(?P<name>[^\W0-9]\w*)?(?P<fmt>.*?)}").unwrap());
+
+static HAS_BRACKETS: Lazy<Regex> = Lazy::new(|| Regex::new(r"\[.*\]").unwrap());
+
+#[derive(Debug)]
+struct FormatFunction {
+    args: Vec<String>,
+    kwargs: HashMap<String, String>,
+    // Whether or not something invalid was found, if it was return immediately
+    invalid: bool,
+}
+
+/// Whether the given string contains characters that are FORBIDDEN in args and
+/// kwargs
+fn contains_invalids(string: &str) -> bool {
+    let invalids = vec!["*", "'", "\"", "await"];
+    for invalid in invalids {
+        if string.contains(invalid) {
+            return true;
+        }
+    }
+    false
+}
+
+impl FormatFunction {
+    fn from_expr(checker: &mut Checker, expr: &Expr) -> Self {
+        let mut final_args: Vec<String> = Vec::new();
+        let mut final_kwargs: HashMap<String, String> = HashMap::new();
+        let mut invalid = false;
+        if let ExprKind::Call { args, keywords, .. } = &expr.node {
+            for arg in args {
+                let arg_range = Range::from_located(arg);
+                let arg_string = checker.locator.slice_source_code_range(&arg_range);
+                invalid = contains_invalids(&arg_string);
+                final_args.push(arg_string.to_string());
+            }
+
+            for keyword in keywords {
+                let KeywordData { arg, value } = &keyword.node;
+                if let Some(key) = arg {
+                    let kwarg_range = Range::from_located(value);
+                    let kwarg_string = checker.locator.slice_source_code_range(&kwarg_range);
+                    invalid = contains_invalids(&kwarg_string);
+                    final_kwargs.insert(key.to_string(), kwarg_string.to_string());
+                }
+            }
+        }
+        Self {
+            args: final_args,
+            kwargs: final_kwargs,
+            invalid,
+        }
+    }
+
+    /// Returns true if args and kwargs are empty
+    fn is_empty(&self) -> bool {
+        self.args.is_empty() && self.kwargs.is_empty()
+    }
+
+    fn consume_arg(&mut self) -> Option<String> {
+        if self.args.is_empty() {
+            None
+        } else {
+            Some(self.args.remove(0))
+        }
+    }
+
+    fn get_kwarg(&self, key: &str) -> Option<String> {
+        self.kwargs.get(key).cloned()
+    }
+
+    /// Returns true if the statement and function call match, and false if not
+    fn check_with_summary(&self, summary: &FormatSummary) -> bool {
```

Comment:
@colin99d - Is this strictly necessary? (Do you know of any case where it would trigger?)